subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -547,6 +516,108 @@ int nrf_cloud_encode_config_response(struct nrf_cloud_data const *const input,
 	return 0;
 }
 
+void nrf_cloud_log_control(int log_level)
+{
+	LOG_DBG("Setting nRF Cloud log level = %d", log_level);
+}
+
+int nrf_cloud_encode_control_response(struct nrf_cloud_data const *const input,
+				      struct nrf_cloud_data *const output,
+				      bool *const has_control)
+{
+	__ASSERT_NO_MSG(output != NULL);
+	__ASSERT_NO_MSG(input != NULL);
+
+	char *buffer = NULL;
+	cJSON *root_obj = NULL;
+	cJSON *desired_obj = NULL;
+	cJSON *reported_obj = NULL;
+	cJSON *state_obj = NULL;
+	cJSON *control_obj = NULL;
+	cJSON *alert_obj = NULL;
+	cJSON *log_obj = NULL;
+	cJSON *input_obj = input ? cJSON_Parse(input->ptr) : NULL;
+
+	if (input_obj == NULL) {
+		return -ESRCH; /* invalid input or no JSON parsed */
+	}
+
+	LOG_DBG("Got delta: %s", (const char *)input->ptr);
+	/* A delta update will have the config inside of state */
+	state_obj = cJSON_DetachItemFromObject(input_obj, JSON_KEY_STATE);
+	control_obj = cJSON_DetachItemFromObject(
+		state_obj ? state_obj : input_obj, JSON_KEY_CTRL);
+	if (control_obj) {
+		LOG_DBG("Shadow delta has control section");
+		alert_obj = cJSON_GetObjectItem(control_obj, JSON_KEY_ALERT);
+		if (cJSON_IsBool(alert_obj)) {
+			nrf_cloud_alert_control(cJSON_IsTrue(alert_obj));
+		} else if (alert_obj != NULL) {
+			LOG_DBG("alertsEn is not a bool");
+		} else {
+			LOG_DBG("alertsEn not found");
+		}
+		log_obj = cJSON_GetObjectItem(control_obj, JSON_KEY_LOG);
+		if (cJSON_IsNumber(log_obj)) {
+			nrf_cloud_log_control((int)cJSON_GetNumberValue(log_obj));
+		}
+	} else {
+		LOG_DBG("Shadow delta does not have control section");
+	}
+	cJSON_Delete(input_obj);
```

Comment:
Since I rewrote this, this comment no longer applies.  If I delete the input_obj too soon, then the child objects I need to access will be deleted too.